libc-top-half/musl/src/env/__init_tls.c

```diff
@@ -16,6 +16,20 @@
 volatile int __thread_list_lock;
 
 #ifndef __wasilibc_unmodified_upstream
+
+extern unsigned char __heap_base;
+extern unsigned char __data_end;
+
+static inline void setup_default_stack_size()
+{
+	ptrdiff_t stack_size = &__heap_base - &__data_end;
```

Comment:
Recent version of wasm-ld will define `__stack_high` and `__stack_low` so we can be more explicit here on recent llvm versions (https://github.com/llvm/llvm-project/blob/4fd0c14a17f91fb7b0d188b509eb3dbdfbfec01a/lld/wasm/Symbols.h#L522-L528.

See https://reviews.llvm.org/D135910.

Perhaps we could declare weak references to them and fall back to  __heap_base/__data_end when they are not found?

Reply:
Great, didn't know about that; I'll update the code.